lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -419,6 +419,32 @@ class DecomposeAtenAddmmOp : public OpRewritePattern<AtenAddmmOp> {
 };
 } // namespace
 
+// Decompose torch.mean into: sum(x)/div(numTensorElements).
+namespace {
+class DecomposeAtenMeanOp : public OpRewritePattern<AtenMeanOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenMeanOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    Value input = op.self();
+    Value output = op.result();
+    BaseTensorType inputTensorType = input.getType().cast<BaseTensorType>();
+    BaseTensorType outputTensorType = output.getType().cast<BaseTensorType>();
+    // AtenDivScalarOp does not support non-floating type input.
+    if (!inputTensorType.getDtype().isa<mlir::FloatType>())
```

Comment:
I don't think we need this check here. In theory, `mean` should work for types other than `float`. If there is currently a temporary limitation in the implementation of `AtenDivScalarOp`, then that op should report an error when that limitation is encountered. `mean` should not have to worry about it. This also prevents the case where `AtenDivScalarOp` is expanded to support more types and yet `mean` still only works for floats because we forgot to update this check.